### PR TITLE
Ylc

### PR DESCRIPTION
**Pull request by yaoliuchang**
实现三个tasks
1. implement logout feature。设置user的状态，检查machine的状态。
2. implement checkOut feature. 
3. implement purchase feature.这个里面实现coins最优配额的算法有问题，结果总是不正确，已经标记为todo，我后续再改。

### CODE REVIEW

File: api/src/main/java/com/mtech/vmcs/controller/CoinController.java [hunk 1]
```diff
@@ -28,10 +31,16 @@ public ResponseEntity<List<Coin>> createCoins(@RequestBody List<Coin> coins) {
     return new ResponseEntity<>(HttpStatus.CREATED);
   }
 
-  // TODO: Wait for later implementation
   @PostMapping("/checkCoin")
-  public ResponseEntity checkCoin(@RequestBody Coin coin) {
-    return null;
+  public ResponseEntity checkCoin(@RequestBody Coin coin) throws IllegalAccessException {
```
Comment (xz1996) [inline api/src/main/java/com/mtech/vmcs/controller/CoinController.java:35]: 1. ```@Autowired private ObjectMapper objectMapper; ``` add it after ```@Autowired private CoinService coinService;```

2. ```suggestion
    public ResponseEntity<ObjectNode> checkCoin(@RequestBody Coin coin) {
    ObjectNode objectNode = objectMapper.valueToTree(coin);
    objectNode.put("isValid", coinService.checkCoin(coin));
    return new ResponseEntity<>(objectNode, HttpStatus.OK);
  }
```

File: api/src/main/java/com/mtech/vmcs/service/impl/CoinServiceImpl.java [hunk 2]
```diff
@@ -35,4 +35,14 @@ public void updateCoins(List<Coin> coins) {
   public void deleteCoins(List<Long> coinIds) {
     coinRepository.deleteAllById(coinIds);
   }
+
+  @Override
+  public Boolean checkCoin(Coin coin){
+    for (Coin everyCoin:
+         getAllCoins()) {
+      if (everyCoin.getValue().equals(coin.getValue()))
```
Comment (xz1996) [inline api/src/main/java/com/mtech/vmcs/service/impl/CoinServiceImpl.java:43]: ```checkCoin``` means validating the weight of coin, not value.

Comment (yaoliuchang) [reply api/src/main/java/com/mtech/vmcs/service/impl/CoinServiceImpl.java:43]: Got it. I will change it later.

File: api/src/main/java/com/mtech/vmcs/controller/CoinController.java (hunk 1 above)
Comment (yaoliuchang) [reply api/src/main/java/com/mtech/vmcs/controller/CoinController.java:38]: Ok, I have done it.

**Review by xz1996 — APPROVED** [review]

**Review by pcdotfan — APPROVED** [review]

**Review by shiyuw09 — APPROVED** [review]

**Pull request merged**